### PR TITLE
vimPlugins: add vim-sneak, unicode.vim, nvim-hs.vim, far.vim, vim-rsi

### PR DESCRIPTION
**Pull request by teto**
~~There is a warning with unicode.vim not being able to write/cache results but it works. Might need some upstream changes.~~

###### Motivation for this change
it's a pain to download unicode data as I have netrw disabled.
vim-sneak is very convenient

~~Waiting for https://github.com/NixOS/nixpkgs/pull/56191 to be able to update plugins,~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by teto** [comment]
I fixed the issue with unicode.vim (the author released a new version that fixed the problem).
The order of the commit is not the best maybe but generated.nix does contain far-vim.
Ready to merge hopefully.

**Review by timokau — CHANGES_REQUESTED** [review]
Would you mind adding the versions the individual plugins are initialized at to the commite (`init at 2019-xx-xx`)? I know its a bit of a nitpick, but its the way we usually do it and shouldn't be much effort with a `git rebase -i`.

While you're at it you could also add one more commit to update all the plugins to the latest versions, replacing  #59343.

Otherwise LGTM.

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -413,4 +416,26 @@ self: super: {
     '';
   });
 
+  vCoolor-vim = super.vCoolor-vim.overrideAttrs(old: {
+    # on linux can use either Zenity or Yad.
+    propagatedBuildInputs = [ gnome3.zenity ];
+    meta = {
+      description = "code-completion for python using python-jedi";
```
Comment (timokau) [inline pkgs/misc/vim-plugins/overrides.nix]: This should start with an uppercase letter

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -413,4 +416,26 @@ self: super: {
...
+      license = stdenv.lib.licenses.mit;
+    };
+  });
+
+  unicode-vim = let
+    unicode-data = fetchurl {
+      url = http://www.unicode.org/Public/UNIDATA/UnicodeData.txt;
+      sha256 = "16b0jzvvzarnlxdvs2izd5ia0ipbd87md143dc6lv6xpdqcs75s9";
+    };
+    in super.unicode-vim.overrideAttrs(old: {
```
Comment (timokau) [inline pkgs/misc/vim-plugins/overrides.nix]: This line is one indentation level too deep

File: pkgs/misc/vim-plugins/overrides.nix [hunk 3]
```diff
@@ -413,4 +416,26 @@ self: super: {
...
+
+      # redirect to /dev/null else changes terminal color
+      buildPhase = ''
+        cp ${unicode-data} autoload/unicode/UnicodeData.txt
```
Comment (timokau) [inline pkgs/misc/vim-plugins/overrides.nix]: Quotes would be nice, although not necessary. Also if we're redirecting to `/dev/null`, we should at least print a status line ("Building unicode cache" or something) to make debugging easier.

**Review by timokau — CHANGES_REQUESTED** [review]
Sorry for sending you through another iteration, but `nix-review` fails:

```
nix build --no-link --keep-going --max-jobs 4 --option build-use-sandbox true -f /home/timo/.cache/nix-review/pr-56503/build.nix
builder for '/nix/store/fpl1jfng1pa0skfjpk2qpy70vwx3i1y0-vimplugin-nvim-hs-vim-2019-01-05.drv' failed with exit code 1; last 10 log lines:
  unpacking sources
  unpacking source archive /nix/store/nkfkjw60szwv9c83ci8311lz26k6ixv9-source
  source root is source
  patching sources
  configuring
  building
  installing
  Building help tags
  Error detected while processing command line:
  E154: Duplicate tag "nvim-hs" in file /nix/store/8hskf9qs40h426hbyxgvxkjrxhlk6nzi-vimplugin-nvim-hs-vim-2019-01-05/share/vim-plugins/nvim-hs-vim/doc/nvim-hs.txtFailed to build help tags!
```

File: pkgs/misc/vim-plugins/overrides.nix (hunk 3 above)
Comment (teto) [reply pkgs/misc/vim-plugins/overrides.nix]: done

**Comment by teto** [comment]
i tried to address your reviews

**Comment by timokau** [comment]
`nix-review` still reports the same build failure of `nvim-hs-vim` though.

**Comment by teto** [comment]
I filed an issue for nvim-hs-vim and removed it from the list. The current darwin failure seems unrelated and linked to a more global darwin failure if I trust the recent issues.

**Review by timokau — APPROVED** [review]

**Comment by timokau** [comment]
Thank you :)

**Comment by teto** [comment]
that's a super cool merge thanks.

**Comment by timokau** [comment]
You're welcome. You are doing good work and are generally responsive to reviews. It's a shame this PR was left to bitrot for so long.

Feel free to ping me in the future if one of your PRs is ready for merge (i.e. at least passes ofBorg and `nix-review`) and you think I'd be qualified to review. No guarantees of course :)

**Pull request merged**